fatal error C1128: number of sections exceeded object file format limit: compile with /bigobj

**Describe the bug**
When compiling the unit tests, I receive the following error:
> HighFive\tests\unit\tests_high_five_multi_dims.cpp : fatal error C1128: number of sections exceeded object file format limit: compile with /bigobj

I recently switch to a newer MSVC version, so that might be related.
MSVC version: 16.10.1+2fd48ab73
CMake version:  3.19.20210118-g54dbc9c

**To Reproduce**
Steps to reproduce the behavior:
1. Build the unit tests

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Version: master

--- a/tests/unit/CMakeLists.txt
+++ b/tests/unit/CMakeLists.txt
@@ -7,6 +7,10 @@ if(NOT Boost_USE_STATIC_LIBS)
   add_definitions(-DBOOST_TEST_DYN_LINK=TRUE)
 endif()
 
+if(MSVC)
+  add_definitions(/bigobj)
+endif()
+
 ## Base tests
 foreach(test_name tests_high_five_base tests_high_five_multi_dims tests_high_five_easy)
   add_executable(${test_name} "${test_name}.cpp")

